pkg/logging/logging.go

```diff
@@ -0,0 +1,126 @@
+package logging
+
+import (
+	"errors"
+	"github.com/go-kit/kit/log"
+	"os"
+)
+
+var componentKey  = "component"
+
+type FilteredLogger struct {
+	logger          log.Logger
+	component       string
+	filterLevel     int
+	currentLogLevel int
+	verbosityLevel  int
+}
+
+func Logger(logger log.Logger) FilteredLogger {
+	defaultLogLevel, ok := logFilterLevels["info"]
+	if !ok {
+		defaultLogLevel = 10
+	}
+	return FilteredLogger{logger: logger,
+		component:   "kubevirt",
+		filterLevel: defaultLogLevel,
+		currentLogLevel: defaultLogLevel,
+	}
+}
+
+type NullLogger struct{}
+
+func (n NullLogger) Log(params ...interface{}) error { return nil }
+
+func New() FilteredLogger {
+	logger := log.NewLogfmtLogger(os.Stderr)
+	return Logger(logger)
+}
+
+var logFilterLevels = map[string]int {
+	"debug":    0,
+	"info":     10,
+	"warning":  20,
+	"error":    30,
+	"critical": 40}
+
+type LogError struct {
+	message string
+}
+
+func (e LogError) Error() string {
+	return e.message
+}
+
+func (l FilteredLogger) Component(comp string) FilteredLogger {
+	l.component = comp
+	return l
+}
+
+func (l FilteredLogger) Log(params ...interface{}) error {
+	if (l.currentLogLevel >= l.filterLevel) {
+		if (l.component != "") {
```

Comment:
that works for me. In fact, doing so will make things much easier: we can use the component as a way to look up/cache the logging object so that it does not need to be re-configured every time it's loaded